src/Sub/Queue/SqsSnsQueue.php

```diff
@@ -2,74 +2,73 @@
 
 namespace PodPoint\AwsPubSub\Sub\Queue;
 
-use Aws\Sqs\SqsClient;
 use Illuminate\Queue\SqsQueue;
 use Illuminate\Support\Arr;
 use Illuminate\Support\Facades\Log;
-use PodPoint\AwsPubSub\Sub\Queue\Jobs\SqsSnsJob;
 
 class SqsSnsQueue extends SqsQueue
 {
     /**
-     * @var array
+     * @inheritDoc
      */
-    protected $events;
+    public function pushRaw($payload, $queue = null, array $options = [])
+    {
+        if ($this->container->bound('log')) {
+            $this->container['log']->error('Unsupported: sqs-sns queue driver is read-only');
+        }
+
+        return null;
+    }
 
     /**
-     * Create a new Amazon SQS SNS subscription queue instance.
-     *
-     * @param  \Aws\Sqs\SqsClient  $sqs
-     * @param  string  $default
-     * @param  string  $prefix
-     * @param  string  $suffix
-     * @param  array  $events
+     * @inheritDoc
      */
-    public function __construct(SqsClient $sqs, $default, $prefix = '', $suffix = '', $events = [])
+    public function later($delay, $job, $data = '', $queue = null)
     {
-        parent::__construct($sqs, $default, $prefix, $suffix);
+        if ($this->container->bound('log')) {
```

Comment:
Later on in this file we call `Log::error` directly - should we update that to match this approach?

Reply:
Yeah I'm using `Log::error()` because it's easier to test with `Log::fake()`.